src/java.base/share/classes/sun/security/tools/keytool/CertAndKeyGen.java

```diff
@@ -81,21 +81,34 @@ public CertAndKeyGen (String keyType, String sigAlg)
         this.keyType = keyType;
     }
 
+    /**
+     * @see #CertAndKeyGen(String, String, String, PrivateKey, X500Name)
+     */
+    public CertAndKeyGen (String keyType, String sigAlg, String providerName)
+    throws NoSuchAlgorithmException, NoSuchProviderException
+    {
+        this(keyType, sigAlg, providerName, null, null);
+    }
+
     /**
      * Creates a CertAndKeyGen object for a particular key type,
-     * signature algorithm, and provider.
+     * signature algorithm, and provider. The newly generated cert will
+     * be signed by the signer's private key when it is provided.
      *
-     * @param keyType type of key, e.g. "RSA", "DSA"
-     * @param sigAlg name of the signature algorithm, e.g. "MD5WithRSA",
-     *          "MD2WithRSA", "SHAwithDSA". If set to null, a default
-     *          algorithm matching the private key will be chosen after
-     *          the first keypair is generated.
+     * @param keyType type of key, e.g. "RSA", "DSA", "X25519", "DH", etc.
+     * @param sigAlg name of the signature algorithm, e.g. "SHA384WithRSA",
+     *          "SHA256withDSA", etc. If set to null, a default
+     *          algorithm matching the private key or signer's private
+     *          key will be chosen after the first keypair is generated.
      * @param providerName name of the provider
+     * @param signerPrivateKey signer's private key
```

Comment:
Add an "(optional)" or "can be null" to the last 2 `@param`s.

Reply:
Done.